planner/core/planbuilder.go

```diff
@@ -723,6 +738,9 @@ func (b *PlanBuilder) getPossibleAccessPaths(indexHints []*ast.IndexHint, tbl ta
 		}
 		for _, idxName := range hint.IndexNames {
 			path := getPathByIndexName(publicPaths, idxName, tblInfo)
+			if !checkReadEngineHasTiKV(b.ctx) {
```

Comment:
There may be multiple warnings here when meeting many index names. We'd better extract the separate logic for isolation check.

Reply:
Addressed.